Language/Classes/Constructors/Factories/return_type_A02_t08.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The return type of a factory whose signature is of the form
+/// factory M or the form factory M.id is M if M is not a generic type;
+/// otherwise the return type is M<T1, . . . , Tn> where T1, . . . , Tn are the
+/// type parameters of the enclosing class.
+///
+/// @description Checks that returning an object whose type is not subtype of
+/// `M` from factory constructor `M` produces a compile error. Test type alias
```

Comment:
```suggestion
/// `M` from a factory constructor `M` produces a compile error. Test type alias
```

(in this case the name is actually the same as the name of the class, so we don't need to make it "declared by `M`")